### PR TITLE
Add formatting providers sample plugin.

### PR DESCRIPTION
**Pull request by akurinnoy**
Signed-off-by: Oleksii Kurinnyi <okurinny@redhat.com>

### Demos:

Document Formatter Demo:

Document Formatter Demo

Document Range Formatter Demo:

Document Range Formatter Demo

On-Type Formatter Demo:

On-Type Formatter Demo

### CODE REVIEW

File: formatter-providers-plugin/src/formatter-providers-frontend-plugin.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+    /**
+     * Register a document formatting provider for all types of files.
+     *
+     * This formatter iterates over each line in the document
+     * and adds the '+' symbol to the beginning of the line.
+     *
+     * Open the Command Palette (press F1 or Ctrl+Shift+P) and run command 'Format Document'
+     */
+    const documentDocumentSelector = {scheme: 'file'};
```
Comment (AndrienkoAleksandr) [inline formatter-providers-plugin/src/formatter-providers-frontend-plugin.ts]: Maybe that's enougth `documentSelector`?

Comment (akurinnoy) [reply formatter-providers-plugin/src/formatter-providers-frontend-plugin.ts]: yes, you're right
There are 3 selectors, so I'll rename them as `documentSector`, `rangeDocumentSelector` and `onTypeDocumentSelector`

File: formatter-providers-plugin/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    output: {
+        filename: 'formatter-providers-frontend-plugin.js',
+        
+        libraryTarget: "var",
+        library: "theia_formatter_providers_plugin",
+                
+        path: path.resolve(__dirname, 'dist')
+    },
+    externals: {
+
```
Comment (mmorhun) [inline formatter-providers-plugin/webpack.config.js]: please correct formatting

File: formatter-providers-plugin/README.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+## formatter-providers-plugin
+
+Simple Theia plugin to demonstrate formatting abilities of Languages API.
+
+It shows how to register formatting providers and provides simple examples of formatters.
+
+### Demo
+
+[![Formatters Demo](http://img.youtube.com/vi/YoYd_nh7Uzo/0.jpg)](https://www.youtube.com/watch?v=YoYd_nh7Uzo)
```
Comment (mmorhun) [inline formatter-providers-plugin/README.md]: I am not sure about links to personal accounts. I would prefer a gif or a few gifs hosted on github

Comment (mmorhun) [reply formatter-providers-plugin/README.md]: I mean a few seconds gifs which demonstrate essentials.

File: formatter-providers-plugin/webpack.config.js (hunk 2 above)
Comment (akurinnoy) [reply formatter-providers-plugin/webpack.config.js]: thanks, I'll fix it

File: formatter-providers-plugin/README.md (hunk 3 above)
Comment (akurinnoy) [reply formatter-providers-plugin/README.md]: good suggestion, I'll add some gifs instead.

**Pull request merged**